### DISCUSSION TITLE
cluster setup- user session not shared on all nodes

### DISCUSSION THREAD

**Discussion by Satishshelake** [keycloak/keycloak#21001] [Archive]
Hello,
   I am checking cluster setup of keycloak using tcpping.  I am issuing access token from one instance and try to access resource which is connected to second instance. It is not working. it gives "401 Unauthorized"  error.

ispn-cache configuration  - 
 
    <jgroups>
        <stack name="tcpping" extends="tcp">
            <TCP bind_port="7802"/>
            <TCPPING
                initial_hosts="127.0.0.1[7801],127.0.0.1[7802]"
                port_range="0"
                stack.combine="REPLACE"
                stack.position="MPING"
                />
        </stack>
    </jgroups>

    <cache-container name="keycloak">
          <transport lock-timeout="60000" stack="tcpping" />
------

keycloak logs shows -- 
INFO  [org.infinispan.CONTAINER] (keycloak-cache-init) ISPN000128: Infinispan version: Infinispan 'Triskaidekaphobia' 13.0.10.Final
INFO  [org.infinispan.CLUSTER] (keycloak-cache-init) ISPN000078: Starting JGroups channel `ISPN` with stack `tcpping`
INFO  [org.infinispan.CLUSTER] (keycloak-cache-init) ISPN000094: Received new cluster view for channel ISPN: [WPF3N9SLR-27814|5] (2) [WPF3N9SLR-27814, WPF3N9SLR-40336]

Please let me know is t working in keycloak 20.0.3 version.

**Comment by Satishshelake** [comment]
I have tried JDBC ping tcpping.  I can see active session on both nodes but If I tries to do authenticate user with resource. It gives '401 Unauthorised'.  Resource server is connected with node1 and I have issued access token from node2.
Do we need to do more configuration for it?

**Reply by Satishshelake** [reply]
The issue was iss field in token
Failed to verify token: Invalid token issuer. Expected 'X' , but was 'Y'.

you can use 
## iss field resolve
hostname-url= <scheme>://<hostname>:<node1port>/<context>

on both nodes.

**Comment by 301lj** [comment]
I have encountered the same issue, have you solved it?
